include/seqan3/alphabet/nucleotide/rna5.hpp

```diff
@@ -56,152 +53,108 @@ namespace seqan3
-    {
-        return value_to_char[static_cast<rank_type>(_value)];
-    }
+private:
+    //!\brief The base class.
+    using base_t = nucleotide_base<rna5, 5>;
 
-    //!\copydoc seqan3::dna4::complement
-    constexpr rna5 complement() const noexcept
-    {
-        return rna5{dna5::complement()};
-    }
-    //!\}
+    //!\brief Befriend nucleotide_base.
+    friend base_t;
+    //!\cond \brief Befriend seqan3::alphabet_base.
+    friend base_t::base_t;
+    //!\endcond
 
-    /*!\name Write functions
+public:
+    /*!\name Constructors, destructor and assignment
      * \{
      */
-    //!\copydoc seqan3::dna4::assign_char
-    constexpr rna5 & assign_char(char_type const c) noexcept
-    {
-        using index_t = std::make_unsigned_t<char_type>;
-        _value = char_to_value[static_cast<index_t>(c)];
-        return *this;
-    }
+    constexpr rna5() : base_t{} {}
+    constexpr rna5(rna5 const &) = default;
+    constexpr rna5(rna5 &&) = default;
+    constexpr rna5 & operator=(rna5 const &) = default;
+    constexpr rna5 & operator=(rna5 &&) = default;
+    ~rna5() = default;
 
-    //!\copydoc seqan3::dna4::assign_rank
-    constexpr rna5 & assign_rank(rank_type const c)
-    {
-        assert(c < value_size);
-        _value = static_cast<internal_type>(c);
-        return *this;
-    }
-    //!\}
+    using base_t::base_t;
 
-    /*!\name Conversion operators
-     * \{
-     */
-    //!\brief Implicit conversion between dna* and rna* of the same size.
-    //!\tparam other_nucl_type The type to convert to; must satisfy seqan3::nucleotide_concept and have the same \link value_size \endlink.
-    template <typename other_nucl_type>
-    //!\cond
-        requires nucleotide_concept<other_nucl_type> && value_size == alphabet_size_v<other_nucl_type>
-    //!\endcond
-    constexpr operator other_nucl_type() const noexcept
+    //!\brief Allow implicit construction from dna/rna of the same size.
+    constexpr rna5(dna5 const & r)
     {
-        return other_nucl_type{_value};
-    }
-
-    //!\brief Explicit conversion to any other nucleotide alphabet (via char representation).
-    //!\tparam other_nucl_type The type to convert to; must satisfy seqan3::nucleotide_concept.
-    template <typename other_nucl_type>
-    //!\cond
-        requires nucleotide_concept<other_nucl_type>
-    //!\endcond
-    explicit constexpr operator other_nucl_type() const noexcept
-    {
-        return detail::convert_through_char_representation<other_nucl_type, std::decay_t<decltype(*this)>>[to_rank()];
+        assign_rank(r.to_rank());
     }
     //!\}
 
-private:
-    //!\copydoc seqan3::dna4::value_to_char
-    static constexpr char_type value_to_char[value_size]
+protected:
+    //!\privatesection
+
+    //!\copydoc seqan3::dna4::rank_to_char
+    static constexpr char_type rank_to_char[value_size]
     {
         'A',
         'C',
         'G',
-        'U',
-        'N'
+        'N',
+        'U'
```

Comment:
You can never rely on the rank, it is implementation defined. But we have changed this consistently now to always be in alphabetical order. This makes debugging a lot easier, because the sorting order is the natural order and also evaluating binary values is much easier.